### PR TITLE
Fix some tests to prepare to drop benefits_intake_uuid from form_submissions

### PR DESCRIPTION
**Pull request by Thrillberg**
## Summary
This PR fixes some tests so that we no longer rely on `FormSubmission.benefits_intake_uuid`, in preparation for [this PR](https://github.com/department-of-veterans-affairs/vets-api/pull/19219).

## Related issue(s)
https://github.com/orgs/department-of-veterans-affairs/projects/1443/views/3?sliceBy%5Bvalue%5D=Thrillberg&pane=issue&itemId=83134444&issue=department-of-veterans-affairs%7CVA.gov-team-forms%7C1636

### CODE REVIEW

File: app/models/form_submission.rb [hunk 1]
```diff
@@ -11,36 +11,16 @@ class FormSubmission < ApplicationRecord
...
-      # otherwise, it falls back to the form submission's original UUID.
-      FormSubmission
-        .select(
-          'DISTINCT form_submissions.id,
-          form_submissions.form_type,
-          COALESCE(latest_attempts.benefits_intake_uuid, form_submissions.benefits_intake_uuid) AS benefits_intake_uuid,
-          form_submissions.created_at'
-        )
-        .left_joins(:form_submission_attempts)
+      select('form_submissions.id, form_submissions.form_type, la.benefits_intake_uuid, form_submissions.created_at')
```
Comment (rhasselle-oddball) [inline app/models/form_submission.rb:15]: do we need any additional tests to be confident this new change has parity with the old version? or the below tests are sufficient?

File: spec/models/form_submission_spec.rb [hunk 2]
```diff
@@ -40,7 +40,7 @@
       it 'returns benefits_intake_id from the form submission' do
         result = FormSubmission.with_latest_benefits_intake_uuid(user_account).with_form_types(['FORM-C']).first
 
-        expect(result.benefits_intake_uuid).to eq(@fsc.benefits_intake_uuid)
+        expect(result.benefits_intake_uuid).to be_nil
```
Comment (rhasselle-oddball) [inline spec/models/form_submission_spec.rb:43]: is this meant to be nil now?  if so, do we need to scan the code for any place that might expect this to exist now?

**Review by rhasselle-oddball — APPROVED** [review]
looks good, couple of questions

File: app/models/form_submission.rb (hunk 1 above)
Comment (Thrillberg) [reply app/models/form_submission.rb:15]: This change was added by @tpharrison and I think the below tests will be sufficient.

File: spec/models/form_submission_spec.rb (hunk 2 above)
Comment (Thrillberg) [reply spec/models/form_submission_spec.rb:43]: Oh good call! This should be `nil` but the `it` block doesn't reflect that.

**Review by pennja — APPROVED** [review]
Looks good, great work! 👍

**Review by RachalCassity — APPROVED** [review]

**Pull request merged**